### PR TITLE
Implement forum sidebar unread

### PR DESCRIPTION
**Pull request by allenwq**
Implemented displaying the unread count on sidebar.

To further display the unread count for each forum on forum index page, need to wait this to be merged: https://github.com/aha-app/calculated_attributes/issues/10

### CODE REVIEW

**Comment by lowjoel** [comment]
I actually implemented this but it requires hat to be merged...

**Comment by lowjoel** [comment]
wait, no, what does this do..?

**Comment by allenwq** [comment]
@lowjoel  It displays the number of unread topics on sidebar

File: app/controllers/components/course/forums_component.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ def settings_sidebar_items
   end
 
   def unread_count
-    # :TODO
-    0
+    return 0 if current_course.nil? || current_user.nil?
```
Comment (lowjoel) [inline app/controllers/components/course/forums_component.rb]: will current_course or current_user be nil? I thought cancancan would have 403'ed the page.

File: app/models/course/forum/topic.rb [hunk 2]
```diff
@@ -99,4 +104,12 @@ def generate_initial_post
   def set_initial_post_title
     posts.first.title = title
   end
+
+  def mark_as_read_for_creator
+    mark_as_read! for: creator
+  end
+
+  def mark_as_read_for_updater
+    mark_as_read! for: updater
```
Comment (lowjoel) [inline app/models/course/forum/topic.rb]: Should have specs for these.

Comment (lowjoel) [inline app/models/course/forum/topic.rb]: Also, do we use the DSL or with ()?

File: app/controllers/components/course/forums_component.rb (hunk 1 above)
Comment (allenwq) [inline app/controllers/components/course/forums_component.rb]: Yah no need, further tested, user will be redirected to the sign in page.

File: app/models/course/forum/topic.rb (hunk 2 above)
Comment (allenwq) [inline app/models/course/forum/topic.rb]: We use without brackets in other places.  

Actually shall we use the suggestion from this style guide ? : https://github.com/airbnb/ruby#no-return-parens

**Pull request merged**